arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java

```diff
@@ -36,187 +38,57 @@
-                return line;
-            }
-        } else if (2 == coords.length) {
-            // This could happen if the Gradle project is making use of the Gradle feature (in preview mode for 4.6+) that has
-            // improved support for BOMs.
-            // In this particular scenario, the dependency should look like the following,
-            // org.wildfly.swarm:jaxrs-multipart -> 2018.5.0-plankton-SNAPSHOT
-
-            StringJoiner joiner = new StringJoiner(":");
-            joiner.add(coords[0]);
-
-            // Extract artifact & version details.
-            String artifactId = coords[1];
-            int idx = artifactId.indexOf(VERSION_UP);
-            if (idx != -1) {
-                String version = artifactId.substring(idx + VERSION_UP.length()).trim();
-                artifactId = artifactId.substring(0, idx).trim();
-                joiner.add(artifactId).add(version);
-                return joiner.toString();
-            }
-        }
-        // Raise an exception if we are unable to determine the dependency format.
-        throw new IllegalArgumentException("Unexpected dependency format: " + line);
-    }
-
-    private String parseLine(String line) {
-        line = line.substring(line.indexOf(PREFIX) + PREFIX.length(), line.length());
-
-        if (line.endsWith(SUFFIX)) {
-            line = line.substring(0, line.indexOf(SUFFIX));
-        }
-        return line;
     }
 
     /**
-     * Scan the project hierarchy (in case of a multi-module project) and return the project that matches the given path
-     * reference.
+     * Retrieve the project dependencies by analyzing the IdeaProject model.
      *
-     * @param project the Gradle project reference.
-     * @param path    the path to match against.
-     * @return the Gradle project reference whose path matches the given input path.
+     * @param connection the Gradle project connection.
+     * @return the project dependencies.
      */
-    private static GradleProject getProjectForDirectory(GradleProject project, Path path) {
-        if (project != null) {
-            if (project.getProjectDirectory().toPath().equals(path)) {
-                return project;
-            }
-            for (GradleProject p : project.getChildren()) {
-                GradleProject searchResult = getProjectForDirectory(p, path);
-                if (searchResult != null) {
-                    return searchResult;
-                }
-            }
-        }
-        return null;
+    private DeclaredDependencies getProjectDependencies(ProjectConnection connection) {
+        // Of the 4 available scopes for a IdeaDependency scope, we ignore "PROVIDED" & "RUNTIME".
+        final List<String> APPLICABLE_SCOPES = Arrays.asList("COMPILE", "TEST");
+        DeclaredDependencies dependencies = new DeclaredDependencies();
+
+        // 1. Get the IdeaProject model from the Gradle connection.
+        IdeaProject prj = connection.getModel(IdeaProject.class);
```

Comment:
Is it guaranteed that there's always an `IdeaProject`, or does that depend on the project having the IntelliJ plugin?

Reply:
I have tested on the following scenarios and it has been working well.

1. All the Gradle projects in the examples repository. None of them have the IntelliJ plugin on them.
2. In my personal project (large multi-module project) which also doesn't have the IntelliJ plugin in it.
3. Trying out different configurations: compile, provided, testCompile.


I need to raise another change that will help improve compatibility with the new [`java-library`](https://docs.gradle.org/current/userguide/java_library_plugin.html) opinion of Gradle. This would support for configurations like `api`, `testImplementation`.

Reply:
Just verified the existing changes with the `java-library` plugin and things are working without any new modifications.